From 45cfa8c37ca70613892ad70cb39b007a484de3dc Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@mainlining.org>
Date: Thu, 14 Aug 2025 05:05:52 +0200
Subject: [PATCH 073/128] arm64: dts: qcom: sm7150: add TRNG node

The sm7150 SoC has a True Random Number Generator, add the node
with the correct compatible set.

Signed-off-by: Jens Reidel <adrian@mainlining.org>
---
 arch/arm64/boot/dts/qcom/sm7150.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm7150.dtsi b/arch/arm64/boot/dts/qcom/sm7150.dtsi
index 0a80ea15f64f..5b4ca20d0dac 100644
--- a/arch/arm64/boot/dts/qcom/sm7150.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm7150.dtsi
@@ -3019,6 +3019,11 @@ gpucc: clock-controller@5090000 {
 			#power-domain-cells = <1>;
 		};
 
+		rng@793000 {
+			compatible = "qcom,sm7150-trng", "qcom,trng";
+			reg = <0x0 0x00793000 0x0 0x1000>;
+		};
+
 		sdhc_1: mmc@7c4000 {
 			compatible = "qcom,sm7150-sdhci",
 				     "qcom,sdhci-msm-v5";
-- 
2.43.0

